Homework day1

GertrudaVal opened the pull request:
Gertruda.
Automation homework.
Could not find how it.only works so left it as comments.

GertrudaVal committed:
tests day 1

GertrudaVal committed:
Test automation homework

DominykasPoskus1 reviewed (changes requested):
Hey, now that's a pretty decent PR. Such a relief to get one that actually runs and finds something(joke)... 

Pros:
1.  Good structure, nice approach, decent tests and coverage. I'm glad you tried covering clear button, only integer checkbox and validation errors. GOOD JOB.
2. There are som functions moved to calcStartPage.js with is also nice.
3. Random numbers makes it more valuable too.
4. Test case titles are descriptive enough to understand, what build is used and what action is done.
5. Kudos for including prototype version into test runs as it helps us to verify that our test exceptions are ok.

Cons:
1. I would suggest to move to 'forEach' instead of 'for' in case we want to select builds by their dropdown option label ("Prototype" for example). Also it will allow us to predefine an array of builds we want to run tests on. This then could be updated to take the build dropdown items list during test run and would no more depend on some hard coded builds you have left in your code.
2. There are some tests that cover certain function, but cannot find the problem (hint: clear button test). Try to run them a couple of times and compare to manual walkthrough. Maybe you could find some improvable areas.
3. Try moving button click, error assertions, extracting calculation result actions to calcStartPage and reuse them throughout the tests. For example, 'await page.inputValue('#numberAnswerField')' could become 'calcStartPage.getAnswer()'.

Good job, I see you put a lot of effort here.

GertrudaVal committed:
Homework updates.

GertrudaVal commented:
Thank you. I added forEach, moved some methods to the start page, and added a test that checks if all the fields exist and are active in the first place.

DominykasPoskus1 commented:
- Integer checkbox becomes a 100% time faulty test, because you input numbers twice. Try to find where.
- For some reason, build 5 tests often fail with timeouts, try to investigate it.

Thanks for doing updates correctly:)

GertrudaVal committed:
Test fixes and an added test.

GertrudaVal commented:
Added test to check multiplication with negative numbers (to check for double negatives, mostly.)
Fixed the bug where the test picked a math function instead of build.

DominykasPoskus1 commented:
Hey, good job. I think I've already requested enough improvements from you :D Thanks for not letting me down and creating some quality tests:)

GertrudaVal committed:
node modules

nodes
